### ISSUE TITLE
Can we remove `ohem` branches in the train net?

### ISSUE THREAD

**Issue by luoyetx** [mahyarnajibi/SSH#17]
the `ohem` layers in `m1@ssh_ohem` shares weight with `m1@ssh` and only provide scores for `anchor_target_layer`, can we remove these layers and use scores from `m1@ssh`. It seems the same.

![tim 20180109233123](https://user-images.githubusercontent.com/2768843/34728648-6cc14ad6-f595-11e7-93b7-1cde6f2f2064.png)

**Comment by loackerc** [comment]
@luoyetx you can remove if you prefer not to use ohem, or you can simply disable it in the configuration file. If you look at the code, ohem is first used to compute a softmax loss, which the probabilities of the classes are then fed into the detection module to sort labels out, hence online hard example mining. Hope I got the concept correct!

**Comment by xuzijian** [comment]
When I read the implementation, I got the same question with @luoyetx. During my own understanding, It seems that we can simplify OHEM branches.

**Comment by foralliance** [comment]
The function of ohem is **online negative and positive mining**

In **anchor_target_layer.py**,  in` # Subsample positives` and` # Subsample negatives` section, you can see its effect.

Of course, I think this is just one way of implementing OHEM, and there must be other ways

**Comment by luoyetx** [comment]
We only need the scores to sort the training samples, the branches added in doesn't backpropagate the gradients, which I think can be removed.
